contrib/node/tests/python/pants_test/contrib/node/tasks/test_apache_thrift_node_gen.py

```diff
@@ -0,0 +1,131 @@
+from textwrap import dedent
+from pants.contrib.node.tasks.apache_thrift_node_gen import ApacheThriftNodeGen
+from pants.contrib.node.targets.node_thrift_library import NodeThriftLibrary
+from pants.contrib.node.targets.node_module import NodeModule
+from pants_test.task_test_base import TaskTestBase
+
+
+class ApacheThriftNodeGenTest(TaskTestBase):
+
+  @classmethod
+  def task_type(cls):
+    return ApacheThriftNodeGen
+
+  def generate_multiple_thrift_targets(self, node_thrift_library):
+    context = self.context(target_roots=[node_thrift_library])
+    apache_thrift_gen = self.create_task(context)
+    apache_thrift_gen.execute()
+
+    def is_synthetic_node_library(target):
+      return isinstance(target, NodeModule) and target.is_synthetic
+
+    synthetic_targets = context.targets(predicate=is_synthetic_node_library)
+
+    self.assertNotEqual(0, len(synthetic_targets))
+    return synthetic_targets[0], synthetic_targets[1]
```

Comment:
Should just return synthetic_targets, otherwise this method is only returning the first two

Reply:
Method was specialized for our tests since we were testing only two targets with one  depending on the other, fixed, now returns the whole list.